installation-manager/src/main/java/org/wildfly/core/instmgr/cli/AddCertificatesCommand.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright The WildFly Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.wildfly.core.instmgr.cli;
+
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.OP;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.RESULT;
+import static org.wildfly.core.instmgr.InstMgrConstants.CERT_FILE;
+
+import java.io.File;
+
+import org.aesh.command.CommandDefinition;
+import org.aesh.command.CommandException;
+import org.aesh.command.CommandResult;
+import org.aesh.command.option.Option;
+import org.aesh.readline.Prompt;
+import org.jboss.as.cli.CommandContext;
+import org.jboss.as.controller.client.Operation;
+import org.jboss.as.controller.client.OperationBuilder;
+import org.jboss.dmr.ModelNode;
+import org.wildfly.core.cli.command.aesh.CLICommandInvocation;
+import org.wildfly.core.instmgr.InstMgrCertificateImportHandler;
+import org.wildfly.core.instmgr.InstMgrCertificateParseHandler;
+import org.wildfly.core.instmgr.InstMgrConstants;
+
+@CommandDefinition(name = "certificates-add", description = "Adds a trusted component certificate.", activator = InstMgrActivator.class)
+public class AddCertificatesCommand extends AbstractInstMgrCommand {
+
+    @Option(name = CERT_FILE, required = true)
+    private File certFile;
+
+    @Option(name = "non-interactive")
+    private boolean nonInteractive;
+
+    public AddCertificatesCommand() {
+    }
+
+    public AddCertificatesCommand(File certFile, boolean nonInteractive) {
+        this.certFile = certFile;
+        this.nonInteractive = nonInteractive;
+    }
+
+    @Override
+    protected Operation buildOperation() {
+        final ModelNode op = new ModelNode();
+        final OperationBuilder operationBuilder = OperationBuilder.create(op);
+
+        op.get(OP).set(InstMgrCertificateImportHandler.DEFINITION.getName());
+        op.get(CERT_FILE).set(0);
+        operationBuilder.addFileAsAttachment(certFile);
```

Comment:
Both, buildParseOperation() and buildOperation() will send the same file as an attachment. A possible improvement would be to save the attachment somehow in the Installation Manager service so it can be referenced by the second operation without sending it again. It would be a minor improvement that will require a method to clean up the attachments. Consider it only if the attachment sizes are big